src/CriticalSection/Driver/PdoMysqlDriver.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace stekycz\CriticalSection\Driver;
+
+use PDO;
+
+/**
+ * @see https://mariadb.com/kb/en/mariadb/get_lock
+ * @see https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_get-lock
+ */
+class PdoMysqlDriver implements IDriver
+{
+	const NO_WAIT = 0;
+
+	/**
+	 * @var PDO
+	 */
+	private $pdo;
+
+	/**
+	 * @var int
+	 */
+	private $lockTimeout;
+
+	public function __construct(PDO $pdo, int $lockTimeout = self::NO_WAIT)
+	{
+		$this->pdo = $pdo;
+		$this->lockTimeout = $lockTimeout;
+	}
+
+	public function acquireLock(string $label) : bool
+	{
+		return $this->runQuery('SELECT GET_LOCK(?, ?)', $label, $this->lockTimeout);
```

Comment:
Sounds right `sha1` should be fine (return 40 chars). I worked primary with MariaDB doc so I miss that.